File: lab3/s1/ejb/src/main/java/ru/uvuv643/ejb/remote/dto/human/request/DeleteHumanBeingByImpactSpeedRequest.java
package ru.uvuv643.ejb.remote.dto.human.request;

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import java.io.Serializable;

@JacksonXmlRootElement(localName = "DeleteHumanBeingByImpactSpeedRequest")
public class DeleteHumanBeingByImpactSpeedRequest implements Serializable {

        @Valid
        @NotNull(message = "Impact speed cannot be null")
        @JacksonXmlProperty(localName = "impactSpeed")
        private Float impactSpeed;

        // Default constructor
        public DeleteHumanBeingByImpactSpeedRequest() {
        }

        // Parameterized constructor
        public DeleteHumanBeingByImpactSpeedRequest(Float impactSpeed) {
                this.impactSpeed = impactSpeed;
        }

        // Getter and Setter
        public Float getImpactSpeed() {
                return impactSpeed;
        }

        public void setImpactSpeed(Float impactSpeed) {
                this.impactSpeed = impactSpeed;
        }
}

File: lab1/s1/src/main/kotlin/ru/uvuv643/soa/api/v1/dto/human/request/DeleteHumanBeingByImpactSpeedRequest.kt
package ru.uvuv643.soa.api.v1.dto.human.request

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull

@JacksonXmlRootElement
data class DeleteHumanBeingByImpactSpeedRequest constructor(

        @field:Valid
        @field:NotNull
        @field:Schema(
                description = "Impact Speed",
                required = true,
        )
        @param:JacksonXmlProperty(namespace = "impactSpeed")
        val impactSpeed: Float?,

        )